scripts/build-docs.js

```diff
@@ -0,0 +1,240 @@
+              let complexID = []
+              i.children.forEach(element => {
+                if (element.value !== undefined) {
+                  complexID += element.value
+                }
+                else {
+                  complexID += element.children[0].value
+                }
+              });
+              elementID = getName(complexID)
+            }
+
+            if (checkDuplicate && elementID === "main-theory") {
+              elementID = "main-theory-1"
+            }
+            if (elementID === "main-theory") {
+              checkDuplicate = true
+            }
+            i.properties = { class: "doc_subtitle", id: elementID }
+          }
+          return i.type !== 'text' || i.value !== '\n'
+        })
+      }
+    ]
+  }
+}
+
+async function readDocs() {
+  let ignore = ['../postcss/docs/README-cn.md', '../postcss/CHANGELOG.md', '../postcss/docs/source-maps.md']
+  let files = (await globby('../postcss/**/**/*.md')).filter(file => !file.match(/node_modules/g))
+  let docs = await Promise.all(
+    files.filter(file => !ignore.includes(file)).map(async file => {
+      let md = await readFile(join(ROOT, file))
+      let tree = await unified()().use(remarkParse).parse(md)
+      tree = await unified()
+        .use(remarkRehype, { allowDangerousHtml: true })
+        .use(rehypeRaw)
+        .use(prepareHTML, file)
+        .use(rehypeHighlight, { prefix: 'code-', ignoreMissing: true })
+        .run(tree)
+      return (tree)
+    })
+  )
+  return docs
+}
+
+async function buildLayout() {
+  let data = await build({
+    mode: 'production',
+    build: {
+      outDir: join(DIST, 'docs'),
+      rollupOptions: {
+        input: join(SRC, 'docs.html')
+      },
+      assetsInlineLimit: 0
+    }
+  })
+  return data.output.find(file => file.fileName === 'docs.html').source
+}
+
+async function makeHTML(tree) {
+  let html = await unified().use(rehypeStringify).stringify(tree)
+  html = tag('section.doc', html)
+  return html
+}
+
+async function saveFile(html, fileName) {
+  let docs = join(DIST, 'docs')
+  if (!existsSync(docs)) await mkdir(docs)
+  let fileTitle
+
+  if (fileName === "postcss-") {
+    fileTitle = 'index.html'
+  }
+  else {
+    fileTitle = fileName + '.html'
+    html = html.replace(/PostCSS Documentation/gm, html.match(/(?<="doc_title">)(.*)(?=<\/h1)/gm))
+  }
+  await writeFile(join(docs, fileTitle), html)
+}
+
+function tag(prefix, attrs, body) {
+  if (typeof body === 'undefined') {
+    body = attrs
+    attrs = {}
+  }
+  let [tagName, cls] = prefix.split('.')
+  if (cls) attrs.class = cls
+  if (Array.isArray(body)) body = body.join('')
+  let attrsString = Object.keys(attrs)
+    .map(i => ` ${i}="${attrs[i]}"`)
+    .join('')
+  return `<${tagName}${attrsString}>${body}</${tagName}>`
+}
+
+function makeSidemenu(contents) {
+  let titles = contents.match(/(?<=<p><a(?:.*?)>)(.*?)(?=<)/gm).filter(title => title !== "").filter(title => title !== "PostCSS and Source Maps")
```

Comment:
Can you format code by Prettier?

This line is definitely longer than we have in code standard

Reply:
Sure, will do now

Reply:
Formatted